### DISCUSSION TITLE
Networking in Corporate Enviroments

### DISCUSSION THREAD

**Discussion by skipmiller** [otya128/winevdm#1339] [Q&A]
Hi. I'd love to use this to allow delivery of old apps in corporate environments using Citrix. Question:  Could an app running via WineVDM access network resources normally? I mainly want to launch Foxpro which connects to data on a 16bit SMB share (I think). Would that work and are there any additional network \ data related work arounds I should consider?

**Comment by cracyc** [comment]
Is it a database file stored on a shared drive, on an sql server on the network or something else?  The first case should work fine if the drive is mounted on the machine running foxpro.  The second should work if the odbc thunk dlls are installed.  I don't know enough about foxpro to know if there are other possibilities that may or may not work.

**Comment by KGHN** [comment]
I have a massive custom FoxPro 2.6 for DOS bookkeeping program with over 30 years of company records in it. No newer Windows will run it as Microsoft has abandoned 16-bit applications, even their own. (I have tried multiple times to convert my program to a newer FoxPro - expensive projects and no luck.)  I'm currently running Win7-32 as the last supporting operating system, but that is end of life and I am concerned about security.  I don't need networking or internet.  Can Winevdm help me?  Is there a Linux Winevdm?

**Reply by cracyc** [reply]
Winevdm won't help you because it's dos support isn't good enough for most things as win16 is the main focus.  You should look at vdos, dosbox-x, msdos-player or msdos in qemu.

**Reply by KGHN** [reply]
Thank you Cracyc, for your good help and advice.  KGHN

**Comment by andvvv** [comment]
Hi, I have experience with old databases. Do you need remote access to database, or just local. I have couple good solutions to run databases like yours. Let me know more details, ty

**Reply by KGHN** [reply]
Thank you, Andvvv,
  My CLBooks is local, single-user use.  As far as I know, I'm the only user and maintainer.
  This program started out in FoxBASE by Fox Software many years ago, updated through Microsoft's FoxPro 2.6 for DOS, the last and best of the DOS compatible FoxPro versions.  Still a 16 bit application.
  I have bought several later FoxPro versions and tried to run/upgrade the application as a 32 bit app.  No luck; the language has changed significantly, upgrading is poorly documented, and Microsoft wants huge fees to help.  The main program is about 328KB of text these days, and I have lots of auxiliary special-purpose programs that use the same database.  These PRG files get compiled into FXP files, and are normally run by the FOXR.EXE runtime.
  My little company has sponsored several useful FoxPro 2.6 for DOS programs written by me.  (I recall that we started with dBASE III+, and also tried the Clipper compiler long ago.)  A solution for this most important custom bookkeeping program need has the potential to trickle down to other programs' users.  At one time we had approx 100 sewer-and-drain-cleaner businesses running our DRAINS software (that IS network and multi-user based.)  It would be nice to revise this and release it to Creative Commons...
  Thank you for your help; I'm looking forward to hearing about your methods.  KGHN

**Comment by KGHN** [comment]
Thank you, Andvvv,
  My CLBooks is local, single-user use.  As far as I know, I'm the only user and maintainer.
  This program started out in FoxBASE by Fox Software many years ago, updated through Microsoft's FoxPro 2.6 for DOS, the last and best of the DOS compatible FoxPro versions.  Still a 16 bit application.
  I have bought several later FoxPro versions and tried to run/upgrade the application as a 32 bit app.  No luck; the language has changed significantly, upgrading is poorly documented, and Microsoft wants huge fees to help.  The main program is about 328KB of text these days, and I have lots of auxiliary special-purpose programs that use the same database.  These PRG files get compiled into FXP files, and are normally run by the FOXR.EXE runtime.
  My little company has sponsored several useful FoxPro 2.6 for DOS programs written by me.  (I recall that we started with dBASE III+, and also tried the Clipper compiler long ago.)  A solution for this most important custom bookkeeping program need thus has the potential to trickle down to other programs' users.  At one time we had approx 100 sewer-and-drain-cleaner businesses running our DRAINS software (that IS network and multi-user based.)  It would be nice to revise this and release it to Creative Commons...
  Thank you for your help; I'm looking forward to hearing about your methods.  KGHN

----- Original Message -----

**Comment by isemoriledesa** [comment]
Has intentado usando maquinas virtuales ?. Tengo programas en foxprox 2.6 que los ejecuto en una maquina virtual windows XP (virutal box)  sobre windows 10.
cuando mencionas red y multisuario. ¿es red local o tambien implica internet ?
